Packs/SysAid/Integrations/SysAid/SysAid.py

```diff
@@ -0,0 +1,877 @@
+        params.update(filters)
+
+        response = self._http_request('GET', 'sr/search', params=params, cookies=self._cookies)
+
+        return response
+
+    def service_record_update_request(self, id_: str, info: List[Dict[str, str]] = None):
+        data = {"id": id_, "info": info}
+
+        response = self._http_request('PUT', f'sr/{id_}', json_data=data, cookies=self._cookies, resp_type='response')
+
+        return response
+
+    def service_record_close_request(self, id_: str, solution: str = None):
+        data = {"solution": solution}
+
+        # 400 - 'This service record is already closed' might be raised - which isn't a real error
+        response = self._http_request('PUT', f'sr/{id_}/close', json_data=data, cookies=self._cookies, resp_type='response',
+                                      ok_codes=(200, 400))
+
+        return response
+
+    def service_record_template_get_request(self, type_: str, fields: str = None, template_id: str = None):
+        params = assign_params(fields=fields, type=type_, template=template_id)
+
+        response = self._http_request('GET', 'sr/template', params=params, cookies=self._cookies)
+
+        return response
+
+    def service_record_create_request(self, type_: str, info: List[Dict[str, str]], fields: str = None,
+                                      template_id: str = None):
+        params = assign_params(fields=fields, type=type_, template=template_id)
+        data = {"info": info}
+
+        response = self._http_request('GET', 'sr/template', params=params, json_data=data, cookies=self._cookies)
+
+        return response
+
+    def service_record_delete_request(self, ids: str, solution: str = None):
+        params = assign_params(ids=ids)
+        data = {"solution": solution}
+
+        # 400 - 'Invalid service record id' might be raised - which isn't a real error
+        response = self._http_request('DELETE', 'sr', params=params, json_data=data, cookies=self._cookies, resp_type='response',
+                                      ok_codes=(200, 400))
+        return response
+
+
+''' HELPER FUNCTIONS '''
+
+
+def create_readable_response(responses: Union[dict, List[dict], str], handle_one_response: Callable, remove_if_null: str = None) \
+        -> Union[str, List[Dict[str, str]]]:
+    """
+    Creates a readable response for responses that have fields in the form of:
+        {
+        'key': 'The Wanted Key',
+        'value': 'The Wanted Value'
+        }
+
+    :param responses: The response to turn to human readable
+    :param handle_one_response: The function to operate on one response to turn it to human readable
+    :param remove_if_null: Field name that if it has no value- the whole response will be ignored
+    """
+    readable_response = []
+
+    if isinstance(responses, dict):
+        responses = [responses]
+
+    if isinstance(responses, str):
+        return responses
+
+    for response in responses:
```

Comment:
Responses might return as a list of dict entries, or as one dict entry.
It might be a string when there is no real data in it and we want to return it this way.

This handles each entry in the response and lets us show the values in it in a human readable way.

What do you mean by "unique to our current scope"? It is used by 6 commands.